html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+When your mouse cursor was away from the button, the button was in the initial state. When you hovered over it you introduced the end state, the hover state, while the transition happened as the color smoothly faded from white to black.
+
+In the above example we animated the CSS `background-color` property. While there are plenty of CSS properties you can animate, you can't animate them all. So keep [this list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties) of "CSS animatable" properties handy so you can check whether or not you can do the transition you want.
+
+Now let's make a basic transition. There are two ways of writing a transition. One is the long form and the other is the short form. The long form looks like this:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition-property: background-color;
+  transition-duration: 2s;
+  transition-timing-function: ease-out;
+  transition-delay: 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The `transition-property` tells us what CSS property will be animated. In this case it is the `background-color` which will change from white to black. The `transition-duration` means the fade from white to black will gradually happen over 2 seconds. The `transition-timing-function` lets us change the speed of the transition over the course of its duration. Here it will `ease-out`, meaning the color change will be faster at the start than at the end of the transition. The `transition-delay` means the transition will start 1 second after the cursor rests on the button. The short form of this is:
+
+~~~css
+button {
+  /* ... other css properties ... */
+  background-color: white;
+  transition: background-color 2s ease-out 1s;
+}
+
+button:hover {
+  background-color: black;
+}
+~~~
+
+The above transition will occur when the user hovers their mouse over the button as indicated by the `:hover` pseudo-class. Different pseudo-classes may be used such as `:active` or `:focus` and you can find the [full list here](https://developer.mozifont-sizella.org/en-US/docs/Web/CSS/Pseudo-classes). We will now explore each of those pseudo-classes one by one.
+
+Just kidding! A skim over that list to familiarize yourself with the CSS pseudo-classes will be enough. While you're there, you absolutely should read the MDN article for [using CSS transitions](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Transitions/Using_CSS_transitions). Make sure to pay attention to the note about the `auto` value and some of the finer points around using transition on multiple properties at the same time. 
```

Comment:
lol if we remove the pseudoclasses above this can go... I'd stick the MDN article in the 'assignment' section.